error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ReactorRules.java

```diff
@@ -1761,6 +1761,51 @@ StepVerifier.FirstStep<? extends T> after(Flux<T> flux) {
     }
   }
 
+  /**
+   * Prefer {@link StepVerifier.LastStep#verify()} over a dangling {@link
+   * StepVerifier#verifyThenAssertThat()}.
+   */
+  static final class StepVerifierVerify {
+    @BeforeTemplate
+    void before(StepVerifier stepVerifier) {
```

Comment:
My intent was indeed to avoid breaking code changes by only matching the "dangling" case in which the created `StepVerifier.Assertions` isn't used or cases we should clearly rewrite.

`StepVerifier#verifyThenAssertThat` has some valid yet niche usages, see [internal usages](https://github.com/search?q=org%3APicnicSupermarket%20verifyThenAssertThat&type=code). Sure, we could suppress the warning here, but IMO the warning would be a false positive. WDYT?

Reply:
Okay - I've given this another thought. Let's proceed as-is, and violations could be suppressed through the warning.  The internal use cases are quite sparse.

Reply:
Discussed offline: I do think @werli has a good point. Ideally we have our cake and eat it too (i.e., have separate "aggressive" and "safe" modes). I'll add a comment describing the current thinking of what that could look like.